### PR TITLE
Preliminary analysis of Park et al. gene set

### PR DESCRIPTION
**Pull request by jjc2718**
**PR description:**

In [Park et al. 2021](https://www.nature.com/articles/s41467-021-27242-3), the authors study interactions between point mutations and CNVs in TCGA data. Essentially, they do an overrepresentation analysis to identify genes/cancer types that have more samples with _both_ a point mutation and a CNV than expected by chance, implying that these genes tend to require "two hits" (the point mutation and the CNV in this case) to be activated (oncogenes) or inactivated (tumor suppressors) in cancer.

Here, we wanted to take the genes/cancer types they identified, and analyze the functional effects in the context of our mutation prediction classifiers. Our hypothesis was that in the "two-hit" genes, samples with "two hits" (a point mutation and a CNV) will have a higher predicted mutation probability than samples with zero or one hit.

This does generally seem to be the case, on average across genes:

image

So this suggests that the samples with "both" a point mutation and a CNV in the gene of interest tend to score higher using our classifiers than samples with "one" of a point mutation or a CNV. Or, in other words, the "functional signature" (in gene expression) of the point mutation/CNV combinations seems to, on average, be "stronger" than the samples with a point mutation or a CNV (or this is my somewhat gratuitous high-level interpretation, anyway).

Next, we want to break this down more by gene/cancer type, which will be a future PR.

**Review notes:**

* `0_download_data/0_download_data.ipynb` is mostly just moved, there's some new code in it to download the gene set from Park et al. but not too many changes
* `0_download_data/1_park_overlap.ipynb` was just a quick check for which new classifiers we needed to train
* Most of the new code is in `3_park_analysis.ipynb`, you can probably focus your review here

### CODE REVIEW

File: 0_download_data/nbconverted/1_park_overlap.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+venn2_unweighted([park_gain_genes, vogelstein_genes], ['Park (gain)', 'Vogelstein'])
+plt.title('Park/Vogelstein gene set overlap')
+
+
+# In[9]:
+
+
+print(len(park_gain_genes - vogelstein_genes))
+print(sorted(park_gain_genes - vogelstein_genes))
+
```
Comment (arielah) [inline 0_download_data/nbconverted/1_park_overlap.py:95]: Maybe add some sort of comment on if this meets your expectations, and how that influences downstream analysis?

File: nbconverted/3_park_analysis.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+copy_thresh_df = (
+    pd.read_csv(Path('data', 'pancan_GISTIC_threshold.tsv'),
+                sep='\t', index_col=0)
+      .drop(columns=['Locus ID', 'Cytoband'])
+)
+copy_thresh_df.columns = copy_thresh_df.columns.str[0:15]
+
+print(copy_thresh_df.shape)
+copy_thresh_df.iloc[:5, :5]
```
Comment (arielah) [inline nbconverted/3_park_analysis.py:93]: Is it weird that sample TCGA-OR-A5J4-01 has CNVs at all five of the alphabetically first genes? Or am I reading this table totally wrong?

File: nbconverted/3_park_analysis.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+# In[12]:
+
+
+copy_gain_df = (copy_thresh_df
+    .replace(to_replace=[-1, -2], value=0)
+    .replace(to_replace=[1, 2], value=1)
+)
+print(copy_gain_df.shape)
+copy_gain_df.iloc[:5, :5]
```
Comment (arielah) [inline nbconverted/3_park_analysis.py:155]: Okay, as the tables go on I'm getting more confused, not less. I think a quick description of what you expect at each step could go a long way. Currently I just look at the table and see some rows with all 0s, some with all 1s, and my mind goes "loading/parsing error"

**Review by arielah — APPROVED** [review]
These are exciting results! I just had a couple points of confusion for clarification. Great work!

File: nbconverted/3_park_analysis.py (hunk 2 above)
Comment (jjc2718) [reply nbconverted/3_park_analysis.py:93]: No, you're reading it correctly. It is a bit weird, but I guess it's plausible that some tumors just have widespread chromosomal instability [e.g. chromoplexy](https://en.wikipedia.org/wiki/Chromoplexy).

We should probably do some basic QC of the CNV data, though - created #4 to do this in the future.

**Pull request merged**